qiskit/pulse/commands/acquire.py

```diff
@@ -68,26 +70,81 @@ def __eq__(self, other):
             return True
         return False
 
-
-class Discriminator(MeasOpts):
-    """Discriminator."""
-
-    def __init__(self, name=None, **params):
-        """Create new discriminator.
-
-        Parameters:
-            name (str): Name of discriminator to be used.
-        """
-        super(Discriminator, self).__init__(name, **params)
-
-
-class Kernel(MeasOpts):
-    """Kernel."""
-
-    def __init__(self, name=None, **params):
-        """Create new kernel.
-
-        Parameters:
-            name (str): Name of kernel to be used.
-        """
-        super(Kernel, self).__init__(name, **params)
+    def __repr__(self):
+        return '%s(%s, duration=%d, kernel=%s, discriminator=%s)' % \
+               (self.__class__.__name__, self.name, self.duration,
+                self.kernel.name, self.discriminator.name)
+
+    def __call__(self,
+                 qubits: Union[Qubit, List[Qubit]],
+                 mem_slots: Union[MemorySlot, List[MemorySlot]],
+                 reg_slots: Union[RegisterSlot, List[RegisterSlot]] = None) -> 'AcquirePulse':
+        return AcquirePulse(self, qubits, mem_slots, reg_slots)
+
+
+class AcquirePulse(Pulse):
```

Comment:
I think there is a lot of reusable logic here that could be done once if this just inherited from a `CommandSchedule`, furthermore an acquire is not a `Pulse`.

Reply:
This will be addressed in another `schedule-rework` PR.